zipkin/src/test/java/zipkin/TestObjects.java

```diff
@@ -78,6 +78,32 @@
           .build()
   ).stream().map(ApplyTimestampAndDuration::apply).collect(toList());
 
+
+  // this object simulates
+  public static final List<Span> TRACEWITHSAMEIDANDSAMEPARENTID = asList(
+    Span.builder().traceId(WEB_SPAN_ID+1).id(WEB_SPAN_ID).name("get")
+      .addAnnotation(Annotation.create(TODAY * 1000, SERVER_RECV, WEB_ENDPOINT))
+      .addAnnotation(Annotation.create((TODAY + 350) * 1000, SERVER_SEND, WEB_ENDPOINT))
+      .build(),
+    Span.builder().traceId(WEB_SPAN_ID+1).parentId(WEB_SPAN_ID).id(WEB_SPAN_ID).name("get")
```

Comment:
when making test data, don't conflate concerns.. for example, it is unclear if linking a span with a bad parent (set to itself) is the problem, or if it is only a problem if somehow duplicate spans with a bad parent are not merged? The annotations added are distracting I think, unless they are contribute to the problem.

start with simplest to break. when you redefine this as a local in the test class, simply mess up the parent id and see if it breaks with a list of only one span.